arbitrator/jit/src/gostack.rs

```diff
@@ -82,30 +121,28 @@ impl GoStack {
 
     pub fn write_u8_ptr(&self, ptr: u32, x: u8) {
         let ptr: WasmPtr<u8> = WasmPtr::new(ptr);
-        ptr.deref(&self.memory).unwrap().set(x);
+        ptr.deref(&self.view()).write(x).unwrap();
     }
 
     pub fn write_u32_ptr(&self, ptr: u32, x: u32) {
         let ptr: WasmPtr<u32> = WasmPtr::new(ptr);
-        ptr.deref(&self.memory).unwrap().set(x);
+        ptr.deref(&self.view()).write(x).unwrap();
     }
 
     pub fn write_u64_ptr(&self, ptr: u32, x: u64) {
         let ptr: WasmPtr<u64> = WasmPtr::new(ptr);
-        ptr.deref(&self.memory).unwrap().set(x);
+        ptr.deref(&self.view()).write(x).unwrap();
     }
 
     pub fn read_slice(&self, ptr: u64, len: u64) -> Vec<u8> {
         let ptr = u32::try_from(ptr).expect("Go pointer not a u32") as usize;
         let len = u32::try_from(len).expect("length isn't a u32") as usize;
-        unsafe { self.memory.data_unchecked()[ptr..ptr + len].to_vec() }
+        unsafe { self.view().data_unchecked()[ptr..][..len].to_vec() }
```

Comment:
I think we can use the safe view read method here now: https://docs.rs/wasmer/latest/wasmer/struct.MemoryView.html#method.read

Reply:
Good call, I've made it safe now